docs/architecture/decisions/0007-non-invasive-read-only-testing.md

```diff
@@ -0,0 +1,31 @@
+# ADR 0007: Non-Invasive, Read-Only Testing
+
+**Date:** 2025-07-25
+
+## Status
+
+Accepted
+
+## Context
+
+See [6. New Test Framework to enable In-Place, Self-Contained Test Execution](./0006-new-test-framework-in-place-self-contained-test-execution.md) for additional context on this decision.
+
+The current Garden Linux test framework modifies the system under test by enabling SSH, installing packages, and making other system-level changes. This approach is suitable for disposable, ephemeral systems but is unsafe for production or long-lived environments, where any mutation can introduce risk, instability, or compliance issues.
+
+Key limitations:
+- **System Mutation:** Tests may alter system configuration, install software, or change files, making the framework unsuitable for production validation.
+- **Operational Risk:** Running tests on production systems risks unintended changes, outages, or security issues.
+- **Limited Applicability:** The framework cannot be safely used on systems that must remain unchanged, such as production servers or persistent VMs.
+
+## Decision
+
+The redesigned test framework will treat the system under test as strictly read-only. Tests must not modify system state, install packages, enable services, or change configuration. The framework itself will not require SSH setup or any other mutation of the target system. All test logic must operate without side effects, ensuring that the system remains unchanged before, during, and after test execution.
```

Comment:
While in principle, yes, we do want to avoid having tests modify the system state; this cannot always be avoided for all tests (e.g. testing of loading kernel modules works). So instead we should build the tests such that tests that modify global system state **MUST** be clearly marked via a pytest marker and skipped unless the test framework is run with an explicit arg to allow modifications.
These tests would then only ever be run on ephemeral targets, such as local test only VMs or during the platform tests.